src/cli/subcommand/restore_shell.rs

```diff
@@ -0,0 +1,80 @@
+use std::{os::unix::prelude::PermissionsExt, process::ExitCode};
+
+use crate::{
+    action::common::ConfigureShellProfile,
+    cli::{ensure_root, CommandExecute},
+    planner::{PlannerError, ShellProfileLocations},
+};
+use clap::{ArgAction, Parser};
+
+/**
+Update the macOS startup files to make Nix usable after system upgrades.
+*/
+#[derive(Debug, Parser)]
+#[command(args_conflicts_with_subcommands = true)]
+pub struct RestoreShell {
+    #[clap(
+        long,
+        env = "NIX_INSTALLER_NO_CONFIRM",
+        action(ArgAction::SetTrue),
+        default_value = "false",
+        global = true
+    )]
+    pub no_confirm: bool,
+
+    #[clap(
+        long,
+        env = "NIX_INSTALLER_EXPLAIN",
+        action(ArgAction::SetTrue),
+        default_value = "false",
+        global = true
+    )]
+    pub explain: bool,
```

Comment:
I don't see `explain` implemented below, can we remove it?

Reply:
My thinking is we'd somehow smush this stuff here into the nice output / runner mechanism we see in `uninstall`. I didn't know how to do that -- or if it was reasonable. I left these in as a reminder to ask -- is that a reasonable / straightforward thing to do?

Reply:
Ahhh like the plan display? We could do that!

Reply:
This turns out to be a bit hard since the associated functions are on `Plan` which requires things like a `planner` which aren't relevant